scripts/intfutil

```diff
@@ -6,74 +6,84 @@ import re
 from tabulate import tabulate
 from natsort import natsorted
 
-
-
 # ========================== Common interface-utils logic ==========================
 
-
 PORT_STATUS_TABLE_PREFIX = "PORT_TABLE:"
+PORT_TRANSCEIVER_TABLE_PREFIX = "TRANSCEIVER_INFO|"
 PORT_LANES_STATUS = "lanes"
 PORT_ALIAS = "alias"
 PORT_OPER_STATUS = "oper_status"
 PORT_ADMIN_STATUS = "admin_status"
 PORT_SPEED = "speed"
 PORT_MTU_STATUS = "mtu"
 PORT_DESCRIPTION = "description"
-
+PORT_OPTICS_TYPE = "type"
 
 def db_connect():
     db = swsssdk.SonicV2Connector(host='127.0.0.1')
     if db == None:
         return None
-
     db.connect(db.APPL_DB)
-
     return db
 
 
 def db_keys_get(db, intf_name):
-
     if intf_name == None:
         db_keys = db.keys(db.APPL_DB, "PORT_TABLE:*")
     elif intf_name.startswith('Ethernet'):
         db_keys = db.keys(db.APPL_DB, "PORT_TABLE:%s" % intf_name)
     else:
         return None
-
     return db_keys
 
 
 def db_port_status_get(db, intf_name, status_type):
     """
     Get the port status
     """
-
     full_table_id = PORT_STATUS_TABLE_PREFIX + intf_name
     status = db.get(db.APPL_DB, full_table_id, status_type)
     if status is None:
         return "N/A"
-
     if status_type == PORT_SPEED and status != "N/A":
        status = '{}G'.format(status[:-3])
-
     return status
 
+def db_connect_optics():
+    """
+    Connect to REDIS STATE DB and get optics info
+    """
+    optics_db = swsssdk.SonicV2Connector(host='127.0.0.1')
+    if optics_db == None:
+        return None
+    optics_db.connect(optics_db.STATE_DB, False)   # Make one attempt only
+    return optics_db
+
+def db_port_optics_get(optics_db, intf_name, type):
```

Comment:
Following the thoughts in my comment above, the first parameter to this function should be named `state_db`, as optical data is not in its own database.

Reply:
Thanks Joe.  I'll update the PR and re-submit.

Reply:
@tsvanduyn: FYI, All you need to do is push the changes to the same branch and the PR will update. You don't need to submit a new PR. If you need assistance, please let me know.

Reply:
Resolved.